azerothcore/azerothcore-wotlk
# feat(Core/Creature): Implement c_t_r & c_t_s

* https://github.com/TrinityCore/TrinityCore/commit/d5fb0a30ec62f3799416326d81d5d14992f9a6df

Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>





## Changes Proposed:
-  Move creature resistance to creature_template_resistance 
-  Move creature spells to creature_template_spell 
-  rename CREATURE_MAX_SPELLS to MAX_CREATURE_SPELLS

## Tests Performed:
- Builds
- Tested in-game as shown below

* creature_template_resistance
![image](https://user-images.githubusercontent.com/24550914/106043194-55989000-60de-11eb-9896-01730f37adb9.png)
![image](https://user-images.githubusercontent.com/24550914/106043226-5d583480-60de-11eb-90e0-02db63bd4f61.png)
![image](https://user-images.githubusercontent.com/24550914/106043259-6943f680-60de-11eb-8440-805f2d9e311f.png)
![image](https://user-images.githubusercontent.com/24550914/106043319-7eb92080-60de-11eb-89d3-62a5453a1c61.png)
![image](https://user-images.githubusercontent.com/24550914/106043455-bcb64480-60de-11eb-86db-3b1b8e39fb00.png)
![image](https://user-images.githubusercontent.com/24550914/106043482-c8097000-60de-11eb-940c-237d7f2b755c.png)

* creature_template_spell
![image](https://user-images.githubusercontent.com/24550914/106043559-e4a5a800-60de-11eb-8829-2b093e73c2ab.png)




## How to Test the Changes:


* creature_template_resistance
```sql
DELETE FROM creature_template_resistance WHERE CreatureID = 299;
INSERT INTO creature_template_resistance (CreatureID, School, Resistance) VALUES
(299, 1, 5000),
(299, 2, 5000),
(299, 3, 5000),
(299, 4, 5000),
(299, 5, 5000),
(299, 6, 5000);
```
1. .go c id 299
1. Test casting spells with the different schools. (HOLY, FIRE, NATURE, FROST, SHADOW, ARCANE)

* creature_template_spell

1. Be ally
2. .npc add temp 28468
3. mount the vehicle and see it has spells



## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
